### PR TITLE
feat: add some tracking

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-8864 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I added some tracking

### CODE REVIEW

**Review by frenautvh — APPROVED** [review]

**Review by marie-j — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Review by Jisay — APPROVED** [review]

File: packages/manager/modules/veeam-enterprise/src/dashboard/license/license.controller.js [hunk 1]
```diff
@@ -10,11 +10,19 @@ export default class VeeamEnterpriseLicenseCtrl {
...
+      `license::${this.action === 'register' ? 'activate' : 'update'}::cancel`,
+    );
     this.goToDashboard();
   }
 
   submitForm(form) {
     if (form.$valid) {
+      this.trackClick(
+        `license::${
+          this.action === 'register' ? 'activate' : 'update'
```
Comment (radireddy) [inline packages/manager/modules/veeam-enterprise/src/dashboard/license/license.controller.js:23]: Nitpick: reuse register and activate strings by extracting them into constants.

**Review by radireddy — APPROVED** [review]

File: packages/manager/modules/veeam-enterprise/src/dashboard/controller.js [hunk 2]
```diff
@@ -9,19 +9,13 @@ export default class VeeamEnterpriseDashboardCtrl {
     CucControllerHelper,
     CucFeatureAvailabilityService,
     VeeamEnterpriseService,
-    goToLicenseActivate,
```
Comment (marie-j) [inline packages/manager/modules/veeam-enterprise/src/dashboard/controller.js:12]: Those are not needed anymore ?

Comment (ghyenne) [reply packages/manager/modules/veeam-enterprise/src/dashboard/controller.js:12]: They are in the binding of the component I created: packages/manager/modules/veeam-enterprise/src/dashboard/component.js
Creating the component was I think the easiest way to use the basic routing tracking.

File: packages/manager/modules/veeam-enterprise/src/dashboard/license/license.controller.js [hunk 3]
```diff
@@ -10,11 +10,19 @@ export default class VeeamEnterpriseLicenseCtrl {
   }
 
   dismissModal() {
+    this.trackClick(
+      `license::${this.action === 'register' ? 'activate' : 'update'}::cancel`,
```
Comment (y4nnL) [inline packages/manager/modules/veeam-enterprise/src/dashboard/license/license.controller.js:14]: Maybe we could avoid code repetition (L14 & L22/L23)
```suggestion
      this.getLicenseTag('cancel'),
```

File: packages/manager/modules/veeam-enterprise/src/details/routing.js [hunk 4]
```diff
@@ -25,15 +25,22 @@ export default /* @ngInject */ ($stateProvider) => {
         goToLicenseTerminate: /* @ngInject */ ($state) => () =>
           $state.go('veeam-enterprise.details.dashboard.license.terminate'),
         breadcrumb: /* @ngInject */ (serviceName) => serviceName,
+        trackingPrefix: () => {
+          return TRACKING_PREFIX;
+        },
+        trackClick: /* @ngInject */ (atInternet, trackingPrefix) => (hit) => {
+          atInternet.trackClick({
+            name: `${trackingPrefix}::${hit}`,
```
Comment (y4nnL) [inline packages/manager/modules/veeam-enterprise/src/details/routing.js:33]: IDK if `trackingPrefix` is injected elsewhere. But if not, no need to make a resolve out of the `TRACKING_PREFIX` constant
```suggestion
        trackClick: /* @ngInject */ (atInternet) => (hit) => {
          atInternet.trackClick({
            name: `${TRACKING_PREFIX}::${hit}`,
```

File: packages/manager/modules/veeam-enterprise/src/dashboard/license/license.controller.js (hunk 3 above)
Comment (ghyenne) [reply packages/manager/modules/veeam-enterprise/src/dashboard/license/license.controller.js]: It's apply

File: packages/manager/modules/veeam-enterprise/src/details/routing.js (hunk 4 above)
Comment (ghyenne) [reply packages/manager/modules/veeam-enterprise/src/details/routing.js]: Indeed it is not necessary, I have modified

File: packages/manager/modules/veeam-enterprise/src/dashboard/license/license.controller.js (hunk 1 above)
Comment (ghyenne) [reply packages/manager/modules/veeam-enterprise/src/dashboard/license/license.controller.js]: I have delegated in a function as suggested in https://github.com/ovh/manager/pull/6765#discussion_r841451303

File: packages/manager/modules/veeam-enterprise/src/dashboard/template.html [hunk 5]
```diff
@@ -70,6 +70,7 @@
                         <oui-action-menu-item
                             aria-label="{{ ::$ctrl.uiActions.manageAutorenew.text }}"
                             data-href="{{ $ctrl.uiActions.manageAutorenew.href }}"
+                            data-on-click="$ctrl.onManageClick()"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/veeam-enterprise/src/dashboard/template.html:73]: to be more clear :
```suggestion
                            data-on-click="$ctrl.onAutorenewManageClick()"
```

File: packages/manager/modules/veeam-enterprise/src/dashboard/controller.js [hunk 6]
```diff
@@ -73,6 +67,10 @@ export default class VeeamEnterpriseDashboardCtrl {
     };
   }
 
+  onManageClick() {
```
Comment (mohammed-zahaf) [inline packages/manager/modules/veeam-enterprise/src/dashboard/controller.js:70]: related to this :
https://github.com/ovh/manager/pull/6765/files#r841654120

**Review by mohammed-zahaf — APPROVED** [review]

File: packages/manager/modules/veeam-enterprise/src/dashboard/template.html (hunk 5 above)
Comment (ghyenne) [reply packages/manager/modules/veeam-enterprise/src/dashboard/template.html]: I applied.

File: packages/manager/modules/veeam-enterprise/src/dashboard/controller.js (hunk 6 above)
Comment (ghyenne) [reply packages/manager/modules/veeam-enterprise/src/dashboard/controller.js]: I applied.

**Review by vks-ovh — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

**Pull request merged**